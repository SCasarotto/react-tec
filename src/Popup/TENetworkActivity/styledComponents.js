import styled from 'styled-components'

import TEPopup from './../TEPopup'
import TESpinner from './../../TESpinner'

export const Popup = styled(TEPopup)`
	${(props) => {
		return `
			display: flex;
			justify-content: center;
			align-items: center;

            & > div{
				background-color: rgba(0,0,0,0.5);
				padding: 60px;
				text-align: center;
            }
        `
	}}
`
export const Spinner = styled(TESpinner)`
	${(props) => {
		return `
            margin-bottom: 30px;
        `
	}}
`
export const Message = styled.p`
	${(props) => {
		const { theme } = props
		return `
			position: relative;
			width: 100%;
			text-align: center;
			font-size: 24px;
			margin: 0px;
			color: ${theme.white};
		`
	}}
`
