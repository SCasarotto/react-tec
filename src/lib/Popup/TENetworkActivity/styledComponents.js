import styled from 'styled-components'

import TEPopup from './../TEPopup'
import TESpinner from './../../TESpinner'

export const Popup = styled(TEPopup)`
	${(props) => {
		return `
            & > div{
				position: absolute;
				top: 50%;
				left: 50%;
				transform: translate(-50%, -50%);
				background-color: rgba(0,0,0,0.5);
				padding: 60px;
				text-align: center;
            }
        `
	}}
`
export const Spinner = styled(TESpinner)`
	${(props) => {
		return `
            margin-bottom: 30px;
        `
	}}
`
export const Message = styled.p`
	${(props) => {
		const { theme } = props
		return `
			position: relative;
			width: 100%;
			text-align: center;
			font-size: 24px;
			color: ${theme.white};
		`
	}}
`
