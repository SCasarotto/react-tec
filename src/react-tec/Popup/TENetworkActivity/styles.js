import { colors } from './../../../config/styles'

export default {
	content : {
		position 			: 'absolute',
	    top 				: '50%',
	    left 				: '50%',
	    WebkitTransform		: 'translate(-50%, -50%)',
	    msTransform			: 'translate(-50%, -50%)',
	    transform 			: 'translate(-50%, -50%)',
		backgroundColor		: 'rgba(0,0,0,0.5)',
		padding 			: '60px',
		textAlign			: 'center',
	},
	spinner : {
	    marginBottom 	: '30px',
	},
	message : {
		position 			: 'relative',
		width 				: '100%',
		textAlign			: 'center',
		fontSize			: '24px',
		color 				: colors.white,
	}
}
