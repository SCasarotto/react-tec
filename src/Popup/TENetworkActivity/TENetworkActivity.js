import React from 'react'
import PropTypes from 'prop-types'

import { Popup, Spinner, Message } from './styledComponents'

const TENetworkActivity = (props) => {
	const { visible, className, message } = props

	return (
		<Popup visible={visible} className={className}>
			<Spinner size='large' className='TENetworkActivitySpinner' />
			{message && <Message className='TENetworkActivityMessage'>{message}</Message>}
		</Popup>
	)
}

TENetworkActivity.propTypes = {
	message: PropTypes.string,
}

export default TENetworkActivity
