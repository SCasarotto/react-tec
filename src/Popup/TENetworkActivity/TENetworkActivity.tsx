import React from 'react'

import { Popup, Spinner, Message } from './styledComponents'
import { TEPopupProps } from './../TEPopup'

export interface TENetworkActivityProps extends TEPopupProps {
	message?: string
}
/**
 *
 * TENetworkActivity is an overlay to indicate some network request or long running process is taking place.
 *
 */
export const TENetworkActivity: React.FC<TENetworkActivityProps> = (props) => {
	const { visible, className = '', message } = props

	return (
		<Popup visible={visible} className={`TENetworkActivity ${className}`}>
			<Spinner size='large' className='TENetworkActivitySpinner' />
			{message && <Message className='TENetworkActivityMessage'>{message}</Message>}
		</Popup>
	)
}
