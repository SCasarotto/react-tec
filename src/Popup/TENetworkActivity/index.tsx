import React from 'react'

import { Popup, Spinner, Message } from './styledComponents'
import { TEPopupProps } from './../TEPopup'

export interface TENetworkActivityProps extends TEPopupProps {
	message?: string
}
export const TENetworkActivity: React.FC<TENetworkActivityProps> = (props) => {
	const { visible, className = '', message } = props

	return (
		<Popup visible={visible} className={`TENetworkActivity ${className}`}>
			<Spinner size='large' className='TENetworkActivitySpinner' />
			{message && <Message className='TENetworkActivityMessage'>{message}</Message>}
		</Popup>
	)
}
