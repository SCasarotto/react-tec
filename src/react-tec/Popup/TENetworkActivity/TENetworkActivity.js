//
//TE Version 0.3.0
//

import React from 'react'
import PropTypes from 'prop-types'

import { Popup, Spinner, Message } from './styledComponents'

const TENetworkActivity = (props) => {
	const { visible, className, message } = props

	return (
		<Popup visible={visible} className={className}>
			{visible && <Spinner size="large" />}
			{message && <Message>{message}</Message>}
		</Popup>
	)
}

TENetworkActivity.propTypes = {
	message: PropTypes.string,
}

export default TENetworkActivity
