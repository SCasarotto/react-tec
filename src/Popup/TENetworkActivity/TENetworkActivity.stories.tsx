import React from 'react'
import { Story, Meta } from '@storybook/react'

import { TENetworkActivity, TENetworkActivityProps } from './TENetworkActivity'

export default {
	title: 'Popup/TENetworkActivity',
	component: TENetworkActivity,
	argTypes: {},
} as Meta

const Template: Story<TENetworkActivityProps> = (args) => <TENetworkActivity {...args} />

export const Default = Template.bind({})
Default.args = {
	message: 'Loading...',
	visible: true,
}
