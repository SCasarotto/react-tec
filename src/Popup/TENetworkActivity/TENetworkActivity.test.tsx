import React from 'react'
import { render } from '@testing-library/react'
import { TENetworkActivity } from './TENetworkActivity'

describe('TENetworkActivity', () => {
	it('matches snapshot', () => {
		const { asFragment } = render(<TENetworkActivity visible />)
		expect(asFragment()).toMatchSnapshot()
	})
})
