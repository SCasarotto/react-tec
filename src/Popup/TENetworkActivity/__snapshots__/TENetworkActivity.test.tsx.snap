// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TENetworkActivity matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnylx kmiImc TEPopup sc-jSFkmK cVtODK TENetworkActivity "
  >
    <div
      class="sc-gtssRu jqgoxm TEPopupContent"
    >
      <div
        class="sc-dlnjPT gPUGaj TESpinner sc-gKAblj bFgUDL TENetworkActivitySpinner"
      >
        <div
          class="sc-hKFyIo vOGXY TESpinnerContainer"
        >
          <div
            class="sc-eCApGN gWcQjK TESpinnerRing TESpinnerRing1"
          />
          <div
            class="sc-eCApGN hqhMHm TESpinnerRing TESpinnerRing2"
          />
          <div
            class="sc-eCApGN iALsOF TESpinnerRing TESpinnerRing3"
          />
          <div
            class="sc-eCApGN cuOCod TESpinnerRing TESpinnerRing4"
          />
          <div
            class="sc-eCApGN ghwANn TESpinnerRing TESpinnerRing5"
          />
          <div
            class="sc-eCApGN beDfwu TESpinnerRing TESpinnerRing6"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
